seo/builder.go

```diff
@@ -0,0 +1,383 @@
+package seo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"reflect"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/qor5/admin/l10n"
+	h "github.com/theplant/htmlgo"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+const (
+	defaultGlobalSEOName = "Global SEO"
+	defaultLocale        = "en"
+)
+
+type (
+	contextVariablesFunc func(interface{}, *Setting, *http.Request) string
```

Comment:
make it public?

Reply:
Okay, making it public can make it easier for users to create context function.